File: ExampleApp/app/src/main/java/com/motiv/example/dao/UsersResponseDao.java
package com.motiv.example.dao;

import androidx.fragment.app.*;
import androidx.room.*;
import com.motiv.example.UsersResponseWithReferences;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

@Dao
public interface UsersResponseDao {

    @Insert
    public long saveUsersResponse(com.motiv.example.UsersResponse usersResponse);

    @Delete
    public int deleteUsersResponse(com.motiv.example.UsersResponse usersResponse);

    @Query("SELECT * FROM usersResponse")
    @Transaction
    public java.util.List<UsersResponseWithReferences> getUsersResponses();

    @Query("SELECT * FROM usersResponse WHERE id = :id")
    @Transaction
    public UsersResponseWithReferences getUsersResponse(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveUsersResponses(java.util.List<com.motiv.example.UsersResponse> usersResponse);
}

File: ExampleApp/app/src/main/java/com/motiv/example/dao/UsersResponseDao.kt
 
package com.motiv.example.dao
import androidx.fragment.app.*
import androidx.room.*
import com.motiv.example.UsersResponse
import com.motiv.example.UsersResponseWithReferences
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import kotlin.collections.List
@Dao
public interface UsersResponseDao {

    @Insert
    public fun saveUsersResponse(usersResponse: com.motiv.example.UsersResponse): Long@Delete
    public fun deleteUsersResponse(usersResponse: com.motiv.example.UsersResponse): Int@Query("SELECT * FROM usersResponse")
    @Transaction
    public fun getUsersResponses(): List<UsersResponseWithReferences>@Query("SELECT * FROM usersResponse WHERE id = :id")
    @Transaction
    public fun getUsersResponse(id: String): UsersResponseWithReferences@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveUsersResponses(usersResponse: List<com.motiv.example.UsersResponse>): Unit
}
